src/main/sensors/initialisation.c

```diff
@@ -457,10 +457,10 @@ static void detectMag(magSensor_e magHardwareToUse)
     magSensor_e magHardware;
 
 #ifdef USE_MAG_HMC5883
-    static hmc5883Config_t *hmc5883Config = 0;
+    hmc5883Config_t *hmc5883Config = 0;
 
 #ifdef NAZE
-    hmc5883Config_t nazeHmc5883Config;
+    static hmc5883Config_t nazeHmc5883Config;
```

Comment:
It would be probably better to use
```
static const hmc5883Config_t nazeHmc5883Config_v1_4 = { ... };
static const hmc5883Config_t nazeHmc5883Config_v5 = { ... };
```
and modify rest of the code (add const to hmc5883lDetect parameter). But that is only cosmetic improvement.